cypress-io/cypress
# chore(deps): refresh dockerode@3.3.1

- Closes #28725

### Additional details

This PR allows [cpu-features](https://www.npmjs.com/package/cpu-features) to be installed on Microsoft Windows operating systems in a Visual Basic environment without error as part of executing `yarn` to install and build Cypress from source. Previously the installation failed with an error that `cmake` could not be found.

The PR removes and re-adds [dockerode](https://www.npmjs.com/package/dockerode) `3.3.1` in [system-tests/package.json](https://github.com/cypress-io/cypress/blob/develop/system-tests/package.json) to force re-evaluation of transient dependencies.

- [cpu-features@0.0.2](https://github.com/mscdex/cpu-features/releases/tag/v0.0.2) is updated to
- [cpu-features@0.0.9](https://github.com/mscdex/cpu-features/releases/tag/v0.0.9)

The newer version of `cpu-features` contains a fix which allows the npm module  [cpu-features](https://www.npmjs.com/package/cpu-features) to rebuild correctly on Windows without generating a message that `cmake` is missing (see commit https://github.com/mscdex/cpu-features/commit/c3704d4c087c74d19f246fffe543814ca4755fd6).

## Steps to test

On Microsoft Windows 11

- Install [Python 3.11](https://devguide.python.org/versions/) from the [Microsoft Store](https://apps.microsoft.com/detail/9NRWMJP3717K?hl=en-us&gl=US). Confirm that `python3 --version` shows `3.11.x`.

- Install [Visual Studio Community](https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=Community) version 2022, using the workload "Desktop development with C++" with its default selections (which includes Windows 11 SDK). Select (or use Visual Studio Installer > Modify) to add Optional component Windows 10 SDK (for example `10.0.20348.0`).

```shell
git clone --branch issue-28725-cpu-features-install https://github.com/MikeMcC399/cypress
cd cypress
```

Use Node.js `18.15.0`

then execute the following

```shell
yarn config set network-timeout 600000
yarn
```

No build error should be reported from `cpu-features`.

There should be no errors between the following two log lines:

```text
[5/6] Building fresh packages...
[6/6] Cleaning modules...
```

_The Yarn network-timeout setting may be necessary as a workaround for a Yarn Classic issue on Windows if Cypress has not previously been installed and there is no Yarn cache built up. Otherwise Yarn may erroneously report_
> info There appears to be trouble with your network connection. Retrying...

followed by
> ESOCKETTIMEDOUT


### How has the user experience changed?

End users see no change. This affects developers only.

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
